examples/credential-registry/src/lib.rs

```diff
@@ -0,0 +1,1969 @@
+#[derive(Serialize, SchemaType, PartialEq, Eq, Clone, Debug)]
+pub struct CredentialData {
+    /// A vector Pedersen commitment to the attributes of the verifiable
+    /// credential.
+    #[concordium(size_length = 2)]
+    commitment:      Vec<u8>,
+    /// A type of the credential that is used to identify which schema the
+    /// credential is based on.
+    credential_type: CredentialType,
+    /// Metadata URL of the credential (not to be confused with the metadata URL
+    /// of the **issuer**).
+    metadata_url:    MetadataUrl,
+}
+
+/// Public data of a verifiable credential.
+#[derive(Serial, DeserialWithState, Debug)]
+#[concordium(state_parameter = "S")]
+pub struct CredentialEntry<S: HasStateApi> {
+    /// If this flag is set to `true` the holder can send a signed message to
+    /// revoke their credential.
+    holder_revocable: bool,
+    /// The date from which the credential is considered valid.
+    valid_from:       Timestamp,
+    /// After this date, the credential becomes expired. `None` corresponds to a
+    /// credential that cannot expire.
+    valid_until:      Option<Timestamp>,
+    /// The nonce is used to avoid replay attacks when checking the holder's
+    /// signature on a revocation message.
+    revocation_nonce: u64,
+    /// Revocation flag
+    revoked:          bool,
+    /// Commitment, schema reference, and credential metadata.
+    /// This data is only needed when credential info is requested. In other
+    /// operations, `StateBox` defers loading credential data.
+    credential_data:  StateBox<CredentialData, S>,
+}
+
+impl<S: HasStateApi> CredentialEntry<S> {
+    /// Compute the credential status based on validity dates and the revocation
+    /// flag. If a VC is revoked the status will be `Revoked` regardless the
+    /// validity dates.
+    fn get_status(&self, now: Timestamp) -> CredentialStatus {
+        if self.revoked {
+            return CredentialStatus::Revoked;
+        }
+        if let Some(valid_until) = self.valid_until {
+            if valid_until < now {
+                return CredentialStatus::Expired;
+            }
+        }
+        if now < self.valid_from {
+            return CredentialStatus::NotActivated;
+        }
+        CredentialStatus::Active
+    }
+
+    fn info(&self, holder_id: CredentialHolderId) -> CredentialInfo {
+        CredentialInfo {
+            holder_id,
+            holder_revocable: self.holder_revocable,
+            commitment: self.credential_data.commitment.clone(),
+            credential_type: self.credential_data.credential_type.clone(),
+            valid_from: self.valid_from,
+            valid_until: self.valid_until,
+            metadata_url: self.credential_data.metadata_url.clone(),
+        }
+    }
+}
+
+/// The registry state.
+#[derive(Serial, DeserialWithState, StateClone)]
+#[concordium(state_parameter = "S")]
+pub struct State<S: HasStateApi> {
+    issuer:              AccountAddress,
```

Comment:
To be consistent, there could be docs comments at every entry in the state struct.

Reply:
Done, added docs to all fields.